recipes-bsp/u-boot/libubootenv_%.bbappend

```diff
@@ -0,0 +1,43 @@
+# Fixup for the libubootenv which rely on uboot-config class for no good reason.
+#
+# This is not intended to be permanent but we need to get the integration
+# working and there is no good solution for now so we are adding this in a
+# non-intrusive way and using the `IMX_DEFAULT_BOOTLOADER` as a guard to do any
+# code execution.
+
+def fixup_uboot_config_dependency(d):
+    ubootmachine = d.getVar("UBOOT_MACHINE")
+    ubootconfig = (d.getVar('UBOOT_CONFIG') or "").split()
+    imx_default_bootloader = d.get('IMX_DEFAULT_BOOTLOADER')
+
```

Comment:
Can we not do similar as for `SPL_BINARY`?

```
UBOOT_MACHINE ?= "${UBOOT_MACHINE_pn-${IMX_DEFAULT_BOOTLOADER}}"
```

Reply:
No; the `uboot-config` fails if both are set ("" is set).